integration/tests/cook/test_basic.py

```diff
@@ -1855,6 +1855,34 @@ def test_basic_docker_job(self):
         job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
         self.assertIn('success', [i['status'] for i in job['instances']])
 
+    @unittest.skipUnless(util.docker_tests_enabled(), "Requires we're in an environment that requires docker images.")
+    @unittest.skipUnless(util.default_submit_pool() is not None, "Test requires a default test pool")
+    def test_default_container_for_pool(self):
+        default_pool = util.default_submit_pool()
+        settings_dict = util.settings(self.cook_url)
+        list_of_pool_match_rules = settings_dict.get("pools", {}).get("default-containers", [])
+        if len(list_of_pool_match_rules) == 0:
+            self.skipTest("No pool match rules defined")
+        matching_containers = [ii["container"] for ii in list_of_pool_match_rules if
+                               re.match(ii["pool-regex"], default_pool)]
+        if len(matching_containers) == 0:
+            self.skipTest("No rule matches the default submit pool")
+        expected_container = matching_containers[0]
+        # Make sure we have a different configured docker image and default image, we fail the test to make sure that we
+        # don't inadvertently skip it.
+        self.failIfEqual(expected_container["docker"]["image"],util.docker_image())
+
+        # Special logic in util.submit_jobs.full_spec maps container=None and removes it from the submitted job spec.
+        job_uuid, resp = util.submit_job(
+            self.cook_url,
+            command='cat /.dockerenv',
+            container=None,
+            max_retries=5)
+        self.assertEqual(resp.status_code, 201)
+        job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
+        self.assertEquals(job["container"]["docker"]["image"], expected_container["docker"]["image"])
```

Comment:
`assertEquals` is deprecated in Python 3:

https://docs.python.org/3/library/unittest.html#deprecated-aliases

Reply:
Good catch.